src/uct/ib/ud/base/ud_inl.h

```diff
@@ -134,6 +134,7 @@ uct_ud_iface_complete_tx_inl(uct_ud_iface_t *iface, uct_ud_ep_t *ep,
                    uct_ud_iface_get_async_time(iface) -
                    ucs_twheel_get_time(&iface->async.slow_timer) +
                    uct_ud_slow_tick());
+    ep->tx.slow_tick = uct_ud_slow_tick();
```

Comment:
let's avoid calling uct_ud_slow_tick() twice in fast path - move this line before ucs_wtimer_add(), and pass  ep->tx.slow_tick to  ucs_wtimer_add(). same for other locations in the code.